yt/frontends/ahf/data_structures.py

```diff
@@ -0,0 +1,156 @@
+"""
+AHF data structures
+
+
+
+"""
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2017, yt Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+import glob
+import os
+import stat
+
+import numpy as np
+
+from yt.data_objects.static_output import \
+    Dataset, \
+    ParticleFile
+from yt.funcs import \
+    setdefaultattr
+from yt.geometry.particle_geometry_handler import \
+    ParticleIndex
+from yt.utilities.cosmology import \
+    Cosmology
+
+from .fields import AHFHalosFieldInfo
+
+
+class AHFHalosFile(ParticleFile):
+    def __init__(self, ds, io, filename, file_id):
+        root, _ = os.path.splitext(filename)
+        candidates = glob.glob(root + '*.AHF_halos')
+        if len(candidates) == 1:
+            filename = candidates[0]
+        else:
+            raise ValueError('Too many AHF_halos files.')
+        self.col_names = self._read_column_names(filename)
+        super(AHFHalosFile, self).__init__(ds, io, filename, file_id)
+
+    def read_data(self):
+        return np.genfromtxt(self.filename, names=self.col_names)
```

Comment:
Instead of reading in the entire file every time, you can use the `usecols` keyword to get only the columns you want.  Since you've got them stored in `self.col_names`, you can add an argument to this function to specify a list of fields and only read those.